frontend/src/scenes/surveys/SurveyView.tsx

```diff
@@ -230,10 +230,11 @@ export function SurveyView({ id }: { id: string }): JSX.Element {
                                                 <div className="mt-6">
                                                     <SurveyAppearance
                                                         type={survey.questions[0].type}
+                                                        surveyQuestionItem={survey.questions[0]}
                                                         appearance={survey.appearance || defaultSurveyAppearance}
                                                         question={survey.questions[0].question}
                                                         description={survey.questions[0].description}
-                                                        link={survey.questions[0].link}
+                                                        link={(survey.questions[0] as LinkSurveyQuestion).link}
```

Comment:
this seems like a bug, there's nothing stopping this survey question from being a different type?

Reply:
no but typescript still gets angery
<img width="769" alt="Screen Shot 2023-07-19 at 10 29 18 AM" src="https://github.com/PostHog/posthog/assets/25164963/10672408-5587-409f-a4c6-30ac982a5147">

Reply:
The way `interface Survey` is currently defined, TypeScript cannot know how the value of `type` is related to the type of `questions` elements. But you can easily get those type system guarantees by refactoring `Survey` the way I refactored `SurveyQuestion` in 333d919897fda025b3f9f6fbcb0680318eef0fe1 (this is called a [discriminating union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions))

Reply:
note: I tried doing this for `SurveyType` and then the kea logicType file didn't like it :( I might've refactored it incorrectly but everything else seemed good except for that so I don't know 🤷‍♀️ .